ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1273,14 +1276,21 @@ public void createTable(Table tbl, boolean ifNotExists,
           principalPrivs.setRolePrivileges(grants.getRoleGrants());
           tTbl.setPrivileges(principalPrivs);
         }
+        boolean createTableUseSuffix = (HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
+            || HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED))
+          && AcidUtils.isTransactionalTable(tbl);
+
+        if (createTableUseSuffix) {
+          tTbl.setTxnId(ss.getTxnMgr().getCurrentTxnId());
```

Comment:
Shall we just push this transactionId for every transactional table?

Reply:
I am relying on txnId a flag that shows if the non-blocking drop is supported. if not set, we won't create a versioned table.

Reply:
Do we have a context object where we can push this info instead of a `magical` flag?
It would be better if we explicitly say "non-blocking" create, instead of depending on the txnId

Reply:
Could it be that we always set the `txnId`, and only set the table property if needed?